Source/WebKit/UIProcess/mac/WKTextAnimationManager.mm

```diff
@@ -95,19 +96,30 @@ - (void)addTextAnimationForAnimationID:(NSUUID *)uuid withData:(const WebKit::Te
         break;
     case WebKit::TextAnimationType::Source:
         effect = adoptNS([PAL::alloc_WTReplaceSourceTextEffectInstance() initWithChunk:chunk.get() effectView:_effectView.get()]);
+        if ([effect respondsToSelector:@selector(setPreCompletion:)]) {
+            effect.get().preCompletion = makeBlockPtr([weakWebView = WeakPtr<WebKit::WebViewImpl>(_webView), nsuuid = uuid] {
```

Comment:
```suggestion
            effect.get().preCompletion = makeBlockPtr([weakWebView = WeakPtr<WebKit::WebViewImpl>(_webView), uuid = retainPtr(uuid)] {
```

I think you need to retain, since nothing else guarantees the lifetime of `NSUUID *`?